SECURITY.md

```diff
@@ -0,0 +1,79 @@
+
+OpenSearch's support for extensions allows for taking already powerful use cases and expanding on them. With this increased functionality comes a larger surface area for misuse, vulnerabilities, and malicious interactions.
+
+By capturing the current state of OpenSearch ecosystem  and the plans for extensions this document outlines several areas for enhancements, features, and practices to incorporate into extensions for OpenSearch.
+
+## Document Phases
+These guidelines and this document are meant to evolve, the follow list captures the different phases this document will undergo.  Some areas might be complete ahead of others. Some areas or items might be marked as invalid/removed using markdown's strike-through.
+
+1. [X] Agreement of areas and 'as-is' state of OpenSearch Plugins and Extensions. **<-- Doc is here**
+2. [ ] All areas have recommendations and areas of investigation are filed as issues and linked back on this document.
+3. [ ] All investigation conclusions are captured and linked in this document, effectively define the scope of work for these areas.  Implementation of work is can be completed or outstanding.
+4. [ ] All planned work has been completed, issues around this work can be completed or outstanding. 
+5. [ ] Document complete, future work and issue will be captured out of band instead of as updates this document.
+
+## Terms
+To keep concepts consistent, this document is using terminology from [NIST Glossary](https://csrc.nist.gov/glossary).
+
+Additional terms:
+* **Plugin** - reference to the existing functionality to extend OpenSearch functionality.
+* **Extension** - reference to the in development functionality to extend OpenSearch.
+
+# Areas
+
+## Host security
+
+Plugins depend on use of the Java Security Manager is use to limit interactions on the host operation system resources (cpu/disk/memory/network/...).  JSM has been deprecated, with its removal scheduled in the next release of the JVM, see [OpenSearch discussion](https://github.com/opensearch-project/OpenSearch/issues/1687). Additional measures are needed to protect system resources.
+
+Extensions are sandboxed from the host system by operating via REST APIs.  This security boundary isolates extensions from executing operation system calls directly on OpenSearch hosts.
+
+## Communications security (COMSEC)
+
+Plugins are loaded into the same java virtual machine instance allowing communicate to OpenSearch through in process java APIs.  Plugins can issue REST API requests to the OpenSearch hosts reusing the standard node-to-node communications, internally called the transport client.
+
+Extensions of OpenSearch communicate via https requests between the nodes on the cluster and the extensions endpoint(s).  This is a bi-direction communication also allows extensions to contact the OpenSearch cluster through its available REST APIs.
+
+## Data Security
+
+OpenSearch stores data in memory and local file system storage.  This data is stored unencrypted.
```

Comment:
If it's unencrypted we should explain how it is presently "secured".

Reply:
The host security is the only check on this data access; we could callout this overlap.  As it occurs to me, the data is not secured in a sufficient manor to justify the use of the term 'secure'.  

Services such as AWS' managed OpenSearch do have at rest encryption for data, and I wanted to make sure that readers of this doc did not conflate the two as I did when I first started writing this doc.

What do you think of leaving this text as it is?